tracing-journald/src/lib.rs

```diff
@@ -121,6 +127,14 @@ impl Subscriber {
         self
     }
 
+    /// Sets the syslog identifier for this logger.
+    ///
+    /// Defaults to the file name of the executable of the current process, if any.
+    pub fn with_syslog_identifier(mut self, identifier: Option<String>) -> Self {
+        self.syslog_identifier = identifier;
```

Comment:
nit, take it or leave it: it might be nice to take `impl Into<Option<String>>` here, so that users can just pass a `String` as well?
```suggestion
    pub fn with_syslog_identifier(mut self, identifier: impl Into<Option<String>>) -> Self {
        self.syslog_identifier = identifier.into();
```

Reply:
> impl Into<Option<String>>

This has the significant drawback that `None` becomes ambiguous and requires type annotation.

Reply:
Hmm, that's a good point, maybe we should leave this as-is.

I guess my follow-up question is, how likely is it that users will ever want to set the syslog identifier to `None`? Should this method just take a `String`, or do we expect a use-case for not having an identifier?

Reply:
Omitting it saves some space, and it's probably not needed if your program doesn't support running outside of a systemd unit.

Reply:
Given that upstream always sets syslog identifier in their C library and that `journalctl -t` is really convenient, I think we should just use `String` here.  I don't think anyone will want to disable it, and most likely won't care I guess.

I'll change it.

Reply:
For the record, I don't have a particularly strong opinion on whether or not we should permit setting it to `None`, I was just wondering what the thought behind it was. :)

Reply:
Uh, I must admit there wasn't much thought behind it 😬😇

Reply:
@hawkw @Ralith I've removed the `Option`, and I've also added a getter for the identifier, in case the application has an interest to get the current identifier.  For instance, the [systemd LogControl interface](https://www.freedesktop.org/software/systemd/man/org.freedesktop.LogControl1.html) has a property which exposes this identifier.